### PR TITLE
Feature/room 

### PR DESCRIPTION
**Pull request by SchoebeP**

### CODE REVIEW

File: controllers/room.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const RoomService = require('../services/room');
+
+class RoomController {
+    static create = async(req,res)=>  {
+        const {name, type, max_people_inside} = req.body;
+        const response = await RoomService.createRoom({ name, type, max_people_inside });
+        return res.status(200).json(response);
```
Comment (mbriand2e) [inline controllers/room.js]: Verify room with this name doesn't exist

File: controllers/room.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        const { id } = req.body;
+        const room = await RoomService.findRoomById(id);
+        if (!room) {
+            return res.status(404).json({ success: false, message: 'Room not found' });
+        }
+
+        if (room.user !== req.user.id || req.user.role !== 'admin') {
+            return res.status(403).json({ success: false, message: 'You are not the owner of this room' });
+        }
+        //const response = await RoomService.deleteRoom(id);
```
Comment (mbriand2e) [inline controllers/room.js]: remove comment

File: schemas/room.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        type: Number,
+        default: 12,
+        validate: {
+            validator: function(v) {
+                return v <= 12;
+            },
+            message: props => `Max people inside should not be more than 12!`
+        }
+    },
+}, { timestamps: { createdAt: 'created_at' } });
```
Comment (mbriand2e) [inline schemas/room.js]:
```suggestion
}, { timestamps: true });
```

File: schemas/user.js [hunk 4]
```diff
@@ -58,7 +58,19 @@ const UserSchema = new mongoose.Schema({
...
+                    return this.role === 'admin';
+                    },
+                toDisplay: function() {
+                    const { __v, _id, password, ...object } = this.toObject();
+                    const jwt = createJsonWebToken(object);
+
+                    return { ...object, token: jwt };
+                    }
+            }
+});
```
Comment (mbriand2e) [inline schemas/user.js]:
```suggestion
}, {
    timestamps: true,
    methods: {
        isAdmin: function() {
            return this.role === 'admin';
        },
        toDisplay: function() {
            const { __v, _id, password, ...object } = this.toObject();
            const jwt = createJsonWebToken(object);

            return { ...object, token: jwt };
        }
    }
});
```

**Review by mbriand2e — CHANGES_REQUESTED** [review]

File: controllers/room.js (hunk 1 above)
Comment (SchoebeP) [reply controllers/room.js]: Done

File: controllers/room.js (hunk 2 above)
Comment (SchoebeP) [reply controllers/room.js]: Done

File: schemas/room.js (hunk 3 above)
Comment (SchoebeP) [reply schemas/room.js]: Done

File: schemas/user.js (hunk 4 above)
Comment (SchoebeP) [reply schemas/user.js]: Done

**Review by mbriand2e — APPROVED** [review]

**Pull request merged**